xdsl/ir/core.py

```diff
@@ -761,6 +761,17 @@ def __iter__(self) -> Iterator[SSAValue]:
     def __len__(self) -> int:
         return len(self._op._operands)  # pyright: ignore[reportPrivateUsage]
 
+    def __eq__(self, other: object):
+        if not isinstance(other, OpOperands):
+            return False
+        return (
+            self._op._operands  # pyright: ignore[reportPrivateUsage]
+            == other._op._operands  # pyright: ignore[reportPrivateUsage]
+        )
+
+    def __hash__(self):
+        return hash(self._op._operands)  # pyright: ignore[reportPrivateUsage]
+
```

Comment:
this would be a fantastic thing to split out into its own PR with tests

Reply:
https://github.com/xdslproject/xdsl/pull/2639